### PR TITLE
Add PR and merged by column to dashboard

### PR DESCRIPTION
**Pull request by samiyaakhtar**
Closes https://github.com/microsoft/bedrock/issues/1166

Adds two new columns to the dashboard, but for configurations where these columns are having no data, these will remain hidden so as to not add empty holes to the dashboard (such as for setups that don't have a PR mechanism into the HLD)

Feedback for this user experience is welcome

Screen Shot 2020-03-23 at 2 43 15 PM

### CODE REVIEW

**Comment by samiyaakhtar** [comment]
Renamed PR Column to "Approval Pull Request"

File: src/client/Dashboard.tsx [hunk 1]
```diff
@@ -904,7 +1075,7 @@ class Dashboard<Props> extends React.Component<Props, IDashboardState> {
         if (queryParams !== "") {
           const promise = HttpHelper.httpGet("/api/author?" + queryParams);
 
-          promise.then(data => {
+          promise.then((data: any) => {
```
Comment (andrebriggs) [inline src/client/Dashboard.tsx]: Out of curiosity why do we need to add `any`?

File: src/client/Dashboard.tsx [hunk 2]
```diff
@@ -895,6 +1012,60 @@ class Dashboard<Props> extends React.Component<Props, IDashboardState> {
...
+    }
+    let str = "";
+    // tslint:disable-next-line: forin
+    for (const key in query) {
+      if (str !== "") {
+        str += "&";
+      }
+      str += key + "=" + encodeURIComponent(query[key]);
+    }
+    return str;
```
Comment (dennisseah) [inline src/client/Dashboard.tsx]: please avoid key in
please use `map` and `join` functions to simplify code. 
```
return Object.keys(query).map(
  k => `${k}=${encodeURIComponent(query[k])}`
).join("&");
```

File: src/client/Dashboard.tsx [hunk 3]
```diff
@@ -895,6 +1012,60 @@ class Dashboard<Props> extends React.Component<Props, IDashboardState> {
...
+
+  private getPRs = () => {
+    try {
+      const state = this.state;
+      this.state.deployments.forEach(deployment => {
+        if (deployment.pr) {
+          const queryParams = this.getPRRequestParams(deployment);
+          if (queryParams !== "") {
+            HttpHelper.httpGet("/api/pr?" + queryParams).then((data: any) => {
+              const pr = data.data as IPullRequest;
```
Comment (dennisseah) [inline src/client/Dashboard.tsx]: please create a new interface and avoid using `any` like
```
interface IQueryResult {
  data: IPullRequest;
}
```

File: src/client/Dashboard.tsx (hunk 1 above)
Comment (samiyaakhtar) [reply src/client/Dashboard.tsx]: I was getting a linting error

File: src/client/Dashboard.tsx (hunk 3 above)
Comment (samiyaakhtar) [reply src/client/Dashboard.tsx]: Removed the use of any at the moment, but the entire repository is using anys in many different places. Will have to create a task to clean those up.

Comment (andrebriggs) [reply src/client/Dashboard.tsx]: @samiyaakhtar let's identify all the places and write it up in a task. The longer we avoid the harder it is to fix

**Comment by samiyaakhtar** [comment]
@dennisseah Let me know if this is good to go

**Comment by dennisseah** [comment]
sorry for the delay

**Pull request merged**